### DISCUSSION TITLE
README badgers

### DISCUSSION THREAD

**Discussion by xuyang2** [emmansun/gmsm#65] [General]
是否可以在 README.md 中加上 goreportcard codecov 的 badger

- https://goreportcard.com/report/github.com/emmansun/gmsm
- https://codecov.io/gh/emmansun/gmsm/branch/main

codecov 好像有一定的接入工作量

**Comment by emmansun** [comment]
好主意

**Reply by xuyang2** [reply]
@emmansun README  中的 codecov badge.svg 链接好像有点问题

https://github.com/codecov/codecov-action/blob/2a829b95deaeea2d11d127cc0358005714ff35ea/README.md?plain=1#L29

**Reply by emmansun** [reply]
具体什么问题？

**Reply by xuyang2** [reply]
没事了
https://github.com/search?q=graph%2Fbadge.svg%3Ftoken+filename%3AREADME.md&type=Code

**Comment by emmansun** [comment]
看来xxx_test这种package还算不出测试覆盖率。

**Reply by xuyang2** [reply]
这种好像要用 `go test -coverpkg=` 单独处理

- https://github.com/microsoft/vscode-go/issues/816#issuecomment-524691721
- https://pkg.go.dev/cmd/go/internal/test
- https://stackoverflow.com/questions/55200883/go-coverage-not-including-functions-in-other-package
- https://go.googlesource.com/proposal/+/master/design/51430-revamp-code-coverage.md
- https://github.com/search?l=YAML&q=go+test+coverpkg&type=Code
- https://medium.com/@taowen/go-test-coverage-for-multi-package-project-d4f36f2b573a

**Reply by xuyang2** [reply]
我在 goland 里简单粗暴加上 `-coverpkg=github.com/emmansun/gmsm/...` 运行覆盖率测试，
算出的 sm4 包的覆盖率比不带 -coverpkg 参数时高：

![image](https://user-images.githubusercontent.com/6643001/178245672-1dd0d20f-4b9e-4254-aa2e-ef44d5202afe.png)

```console
$ git rev-parse --verify HEAD
f51e5cca1f13d0490fd882c0574ebc7a7eaa31bb
```

**Reply by emmansun** [reply]
加上了-coverpkg=./...
